### PR TITLE
Move include/bbcode to src/ part 1 : Everything but bbcode()

### PR DESCRIPTION
**Pull request by MrPetovan**
Part of #3878

Exactly as the name says.

I'm not terribly happy about the name conflict between `Util\Network` and `Core\Network` but I haven't found a good name for `Util\Network` yet. Suggestions welcome.

### CODE REVIEW

File: src/Content/Text/BBCode.php [hunk 1]
```diff
@@ -636,4 +646,1341 @@ public static function limitBodySize($body)
...
+			$posted = $matches[1];
+		}
+
+		preg_match('/posted="(.*?)"/ism', $attributes, $matches);
+		if (x($matches, 1)) {
+			$posted = $matches[1];
+		}
+
+		// We only call this so that a previously unknown contact can be added.
+		// This is important for the function "get_contact_details_by_url".
```
Comment (tobiasd) [inline src/Content/Text/BBCode.php]: The `get_contact...` function now has a different name, shoud be updated in the comment as well.

File: src/Content/Text/BBCode.php [hunk 2]
```diff
@@ -636,4 +646,1341 @@ public static function limitBodySize($body)
...
+		$return = preg_replace_callback("&\[url=([^\[\]]*)\]\[img\](.*)\[\/img\]\[\/url\]&Usi", 'self::cleanPictureLinksCallback', $text);
+		return $return;
+	}
+
+	private static function textHighlightCallback($match)
+	{
+		if (in_array(strtolower($match[1]),
+				['php', 'css', 'mysql', 'sql', 'abap', 'diff', 'html', 'perl', 'ruby',
+				'vbscript', 'avrc', 'dtd', 'java', 'xml', 'cpp', 'python', 'javascript', 'js', 'sh'])
+		) {
```
Comment (tobiasd) [inline src/Content/Text/BBCode.php]: This is the list of languages for which the syntax highlight works right? Could you add it to the BBCode documentation to the `[code]` tag reference.

File: src/Content/Text/BBCode.php [hunk 3]
```diff
@@ -636,4 +646,1341 @@ public static function limitBodySize($body)
...
+				$return = OEmbed::getHTML($url, $title);
+			} catch (Exception $ex) {
+				$return = $match[0];
+			}
+
+			return $return;
+		};
+
+		// Hide all [noparse] contained bbtags by spacefying them
+		// POSSIBLE BUG --> Will the 'preg' functions crash if there's an embedded image?
```
Comment (tobiasd) [inline src/Content/Text/BBCode.php]: Is there a Daxygen identifier for a *possible bug*? Just wondering if this reminder could be more prominent or if this is our usual way to mark these places.

Comment (MrPetovan) [reply src/Content/Text/BBCode.php]: This is our usual way of burying potential issues without signing, which means nobody can know who wrote this after such a move.

File: src/Content/Text/BBCode.php (hunk 2 above)
Comment (MrPetovan) [reply src/Content/Text/BBCode.php]: Done.

File: src/Content/Text/BBCode.php (hunk 1 above)
Comment (MrPetovan) [reply src/Content/Text/BBCode.php]: Done.

**Comment by annando** [comment]
Concerning this "network": Would possibly "protocol" be the better name?

**Comment by MrPetovan** [comment]
I hesitated between both, then I stuck with `Network` to match the constants name, but it certainly would solve the conflict.

**Comment by annando** [comment]
I would prefer the protocols since it fits much better into what it will tell us: That we have to make different coding because of the different protocols.

**Comment by MrPetovan** [comment]
I just noticed the `PROTOCOL_*` constants in `boot.php`, what should we do with them regarding to the naming?

**Comment by annando** [comment]
So move it.

**Comment by MrPetovan** [comment]
I left placeholders, I'll do the final move when I'll remove `boot.php`.

**Comment by tobiasd** [comment]
> This is our usual way of burying potential issues without signing,
> which means nobody can know who wrote this after such a move.

Ok :-D

**Comment by tobiasd** [comment]
> Done.

Thanks!

**Comment by tobiasd** [comment]
@annando can this be merged?

**Comment by annando** [comment]
I hadn't looked deeper since I thought that you would merge it.

**Comment by tobiasd** [comment]
Then the misunderstanding was on my side. Though you wanted to check the "move it" stuff.

**Comment by annando** [comment]
I have one wish for the second part: Could you create some functions like "toTwitter" or "forDiaspora" in the BBCode class that are calling the "convert" function with the appropriate parameters?

I think that would be much more readable than this crazy ```$simple_html``` parameter and the other parameters that mostly doesn't seem to serve any useful purpose.

**Comment by MrPetovan** [comment]
Should I be using hooks or are we expected to send text to Twitter even without the Addon?

**Comment by annando** [comment]
The different values in "simple_html" are directly connected to the different output formats. So I don't see a reason why we shouldn't name it after them directly in the class.

**Comment by MrPetovan** [comment]
If a specific value of simplehtml is only used from an addon, then we can and we should move the specific code in the addon.

**Comment by annando** [comment]
Have a look inside the ```convert``` function. It cannot be isolated, I think.

**Pull request merged**